### ISSUE TITLE
Allow the Deferred returned by twisted.internet.task.deferLater to be cancelled

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4318]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4318|
|Type|enhancement|
|Created|2010-02-25 01:43:18Z|
|Branch|https://github.com/twisted/twisted/tree/cancel-deferlater-4318|

Cancelling a `Deferred` returned from `deferLater` should fail the `Deferred` with `CancelledError` and prevent the call to the function passed to `deferLater` from actually happening.

<details><summary>Searchable metadata</summary>

```
trac-id__4318 4318
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_cancel_deferlater_4318 branches/cancel-deferlater-4318
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1267062198000000 1267062198000000
changetime__1267383174000000 1267383174000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [28525]) Branching to 'cancel-deferlater-4318'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Looks good. The branch is just missing a NEWS file. +1!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [28580]) Merge cancel-deferlater-4318

Author: exarkun
Reviewer: therve
Fixes: [#4318](https://github.com/twisted/twisted/issues/4318)

Implement a canceller function for the `Deferred` returned by `deferLater` which
cancels the underlying `DelayedCall`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
